Kotlin stackdriver debugger support

etanshaul opened the pull request:
fixes #2253 

Supports Kotlin file types as breakpoint sources. 

- Kotlin and Java language specific logic lives in optionally included plugin.xml's: `stackdriver-debugger-java.xml` and `stackdriver-debugger-kotlin.xml`
  - Currently all that is needed is a simple language-specific implementation of the platform `JavaDebugAware` extension point which probably would have been better named as `JvmDebugAware` 

- Since the language specific code was so minimal I decided not to create new kotlin and java modules under stackdriver and instead keep all the source in the stackdriver module. However, the kotlin plugin is still optionally included in the `google-cloud-tools-plugin` module's plugin.xml - which means things will still work for Java if the user doesn't have the Kotlin plugin installed. Of course this now has the downside that the build time classpath doesn't match the runtime completely, but I think its OK given the extent of the language specific support.

ivanporty commented on stackdriver-debugger/src/com/google/cloud/tools/intellij/stackdriver/debugger/KotlinLangDebugAware.java:
```diff
@@ -0,0 +1,30 @@
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.kotlin.idea.KotlinFileType;
+
+public class KotlinLangDebugAware extends JavaDebugAware {
```
It looks like javadoc would be helpful here.

ivanporty commented on stackdriver-debugger/src/com/google/cloud/tools/intellij/stackdriver/debugger/JavaLangDebugAware.java:
```diff
@@ -0,0 +1,30 @@
+import com.intellij.psi.PsiFile;
+import org.jetbrains.annotations.NotNull;
+
+public class JavaLangDebugAware extends JavaDebugAware {
```
Same, javadoc would be helpful here.

ivanporty commented on stackdriver-debugger/src/com/google/cloud/tools/intellij/stackdriver/debugger/CloudBreakpointHandler.java:
```diff
@@ -353,13 +354,15 @@ public void registerBreakpoint(
     }
 
     PsiFile javaFile = psiManager.findFile(ideBreakpoint.getSourcePosition().getFile());
```
Since this is no longer a java file only, should we name it a bit more broad?

etanshaul replied:
definitely

etanshaul commented:
updated based on comments so far

etanshaul commented on google-cloud-tools-plugin/resources/META-INF/plugin.xml:
```diff
@@ -51,6 +52,7 @@
   <xi:include href="/META-INF/google-cloud-storage.xml" xpointer="xpointer(/idea-plugin/*)"/>
   <xi:include href="/META-INF/google-cloud-repos.xml" xpointer="xpointer(/idea-plugin/*)"/>
   <xi:include href="/META-INF/google-cloud-apis.xml" xpointer="xpointer(/idea-plugin/*)"/>
+  <xi:include href="/META-INF/stackdriver-debugger.xml" xpointer="xpointer(/idea-plugin/*)"/>
```
My initial inclination is to move this to live under `google-cloud-java.xml` which presumably will be available for all JVM languages. And perhaps collapse `stackdriver-debugger.xml` and `stackdriver-debugger-java.xml` into one.

But I need to revisit this.

etanshaul commented:
@ivanporty I fixed your comment about stackdriver being available for other IDEs (like PyCharm). See me last commit: I moved the stackdriver plugin xml to live under `google-cloud-java.xml`

Now, debugger is only available when the Java module is available (IDEA). Kotlin debugging still works.
